### PR TITLE
CRD documentation list all versions

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- create one section per crd version
- create one table per spec and status
- add "type" column with indicator if mandatory

**Related issue(s)**

### CODE REVIEW

File: docs/05-technical-reference/00-custom-resources/telemetry-03-tracepipeline.md [hunk 1]
```diff
@@ -43,44 +43,56 @@ For details, see the [TracePipeline specification file](https://github.com/kyma-
 <!-- The content between "TABLE-START" and "TABLE-END" will be replaced -->
 
 <!-- TABLE-START -->
+### TracePipeline.v1alpha1.telemetry.kyma-project.io
+**Spec:**
+
+<!-- TracePipeline v1alpha1 telemetry.kyma-project.io -->
+| Parameter         | Type | Description                                   |
+| ------------------| ---- | --------------------------------------------- |
+| **spec.output** | object | Defines a destination for shipping trace data. Only one can be defined per pipeline. |
```
Comment (a-thaler) [inline docs/05-technical-reference/00-custom-resources/telemetry-03-tracepipeline.md:52]: Can you remove the "spec." prefix, it is redundant and will reduce the width of the column

**Comment by a-thaler** [comment]
Could you ad the "required" flag behind the type so that we have covered all requirements already. That would be great :)

File: docs/05-technical-reference/00-custom-resources/telemetry-03-tracepipeline.md (hunk 1 above)
Comment (k15r) [reply docs/05-technical-reference/00-custom-resources/telemetry-03-tracepipeline.md:52]: sure

**Comment by k15r** [comment]
> Could you ad the "required" flag behind the type so that we have covered all requirements already. That would be great :)

let me try. but as this tool uses a rather basic way to "parse" openapi it will not be nice

File: docs/05-technical-reference/00-custom-resources/evnt-01-subscription.md [hunk 2]
```diff
@@ -35,15 +35,123 @@ spec:
...
+| **types.originalType** | string |  |
+
+### Subscription.v1alpha1.eventing.kyma-project.io
+
+**Spec:**
+
+<!-- Subscription v1alpha1 eventing.kyma-project.io -->
+| Parameter         | Type | Description                                   |
+| ------------------| ---- | --------------------------------------------- |
+| **** | object | SubscriptionSpec defines the desired state of Subscription. |
```
Comment (a-thaler) [inline docs/05-technical-reference/00-custom-resources/evnt-01-subscription.md:96]: The "spec" root attribute need to be excluded from generation

File: docs/05-technical-reference/00-custom-resources/telemetry-01-logpipeline.md [hunk 3]
```diff
@@ -68,76 +68,91 @@ For details, see the [LogPipeline specification file](https://github.com/kyma-pr
...
-| **status.conditions** | An array of conditions describing the status of the pipeline. |
-| **status.conditions.lastTransitionTime** | An array of conditions describing the status of the pipeline. |
-| **status.conditions.reason** | An array of conditions describing the status of the pipeline. |
-| **status.conditions.type** | The possible transition types are:<br>- `Running`: The instance is ready and usable.<br>- `Pending`: The pipeline is being activated. |
-| **status.unsupportedMode** | Is active when the LogPipeline uses a `custom` output or filter; see [unsupported mode](./../../01-overview/main-areas/telemetry/telemetry-02-logs.md#unsupported-mode#unsupported-mode). |<!-- TABLE-END -->
\ No newline at end of file
+| Parameter         | Type | Description                                   |
+| ------------------| ---- | --------------------------------------------- |
+| **** | object | Defines the desired state of LogPipeline |
+| **files** | Provides file content to be consumed by a LogPipeline configuration | files |
```
Comment (a-thaler) [inline docs/05-technical-reference/00-custom-resources/telemetry-01-logpipeline.md:79]: The description is in second column...

**Review by pbochynski — APPROVED** [review]

**Review by Teneroy — CHANGES_REQUESTED** [review]

File: hack/table-gen/table-gen_test.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_flatten(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (k15r) [reply hack/table-gen/table-gen_test.go:8]: honestly, im not really sure. That name was generated automatically by goland. 
I see people using Test_XX and TestXX. 
Changed it now to TestFlatten ...

**Review by mmitoraj — COMMENTED** [review]
Hi, nice new features :) I have a few suggestions:
- Please remove all parameters' names from the **Description** to avoid duplication. For example, instead of:
```
| **backend.emshash** | integer | Emshash defines the hash for the Subscription in EventType. |
```
write:
```
| **backend.emshash** | integer | Defines the hash for the Subscription in EventType. |
```
- For me, the information if something is **required** doesn't look good in the **Type** column. How about adding it in parentheses in a new line after the parameter name in the **Parameter** column? For example:
```
 | **protocolsettings.webhookAuth.tokenUrl** (required) | string | TokenURL defines token URL for OAuth2 |
```
It should go to a new line under the parameter name, right?
- I added a few suggestions to the descriptions themselves but is there a point of doing that in the autogenerated document? Can I review those in the source files?

File: docs/05-technical-reference/00-custom-resources/evnt-01-subscription.md [hunk 5]
```diff
@@ -35,15 +35,114 @@ spec:
...
+**Spec:**
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **config** | map[string]string | Config defines the configurations that can be applied to the eventing backend. |
+| **id** | string | ID is the unique identifier of Subscription, read-only. |
+| **sink** | string **required** | Sink defines endpoint of the subscriber |
+| **source** | string **required** | Source Defines the source of the event originated from. |
+| **typeMatching** | string | TypeMatching defines the type of matching to be done for the event types. |
+| **types** | []string | Types defines the list of event names for the topics we need to subscribe for messages. |
```
[the remark this answers is not in the document]
Comment (k15r) [reply docs/05-technical-reference/00-custom-resources/evnt-01-subscription.md]: yes it is

File: docs/05-technical-reference/00-custom-resources/telemetry-02-logparser.md [hunk 6]
```diff
@@ -34,11 +34,21 @@ For details, see the [LogParser specification file](https://github.com/kyma-proj
...
-| **status.conditions.lastTransitionTime** | An array of conditions describing the status of the parser. |
-| **status.conditions.reason** | An array of conditions describing the status of the parser. |
-| **status.conditions.type** | The possible transition types are:<br>- `Running`: The parser is ready and usable.<br>- `Pending`: The parser is being activated. |<!-- TABLE-END -->
+### LogParser.telemetry.kyma-project.io/v1alpha1
+
+**Spec:**
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **parser**  | string | \[Fluent Bit Parsers\]\(https://docs\.fluentbit\.io/manual/pipeline/parsers\)\. The parser specified here has no effect until it is referenced by a \[Pod annotation\]\(https://docs\.fluentbit\.io/manual/pipeline/filters/kubernetes\#kubernetes\-annotations\) on your workload or by a \[Parser Filter\]\(https://docs\.fluentbit\.io/manual/pipeline/filters/parser\) defined in a pipeline's filters section\. |
```
Comment (mmitoraj) [inline docs/05-technical-reference/00-custom-resources/telemetry-02-logparser.md:43]: Links don't display correctly throughout all the documents.

File: docs/05-technical-reference/00-custom-resources/telemetry-02-logparser.md [hunk 7]
```diff
@@ -34,11 +34,21 @@ For details, see the [LogParser specification file](https://github.com/kyma-proj
...
+
+**Status:**
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **conditions**  | array | An array of conditions describing the status of the parser\. |
+| **conditions.lastTransitionTime**  | string | An array of conditions describing the status of the parser\. |
+| **conditions.reason**  | string | An array of conditions describing the status of the parser\. |
+| **conditions.type**  | string | The possible transition types are:\<br\>\- \`Running\`: The parser is ready and usable\.\<br\>\- \`Pending\`: The parser is being activated\. |
```
Comment (mmitoraj) [inline docs/05-technical-reference/00-custom-resources/telemetry-02-logparser.md:52]: Formatting in the **Description** column doesn't display correctly throughout all the documents.

**Review by mmitoraj — COMMENTED** [review]
In the **Description** column links don’t display correctly and the formatting is broken (code font, bold, etc.).

**Review by Teneroy — DISMISSED** [review]

**Review by mmitoraj — APPROVED** [review]

**Review by mmitoraj — APPROVED** [review]

**Review by pbochynski — APPROVED** [review]

**Pull request merged**